src/archivematicaCommon/lib/workflow.py

```diff
@@ -0,0 +1,66 @@
+import sys
+
+import grpc
+
+from protos import workflow_pb2
+
+
+def _rendezvous_exception(func):
+    def _decorated(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except grpc._channel._Rendezvous as e:
+            if e.code() == grpc.StatusCode.UNAVAILABLE:
+                raise ServerUnavailableError(e.code())
+    return _decorated
+
+
+class Error(Exception):
+    def __init__(self, message):
+        self.message = 'Workflow client error: ' + message
+
+
+class ServerUnavailableError(Error):
+    def __init__(self, code):
+        self.code = code
+        self.message = 'server unavailable, code ' + self.code
+
+
+class Client(object):
+    DEFAULT_ADDRESS = 'localhost:50050'
+
+    def __init__(self, addr=None):
+        self.addr = addr if addr is not None else self.DEFAULT_ADDRESS
+
+        self._init_metadata()
+
+        self.channel = grpc.insecure_channel(self.addr)
```

Comment:
Yes to this but I'm hoping I can simplify that as much as possible. If we adopt the sidecar model (i.e. we run something like linkerd in every host) we could just use the insecure channel to speak to the sidecar and let it take care of the encryption for us. See this article: [transparent TLS with linkerd](https://blog.buoyant.io/2016/03/24/transparent-tls-with-linkerd/). That would make things easier from the perspective of our microservices but also operationally speaking. E.g. we would only have to set up the certificates in the sidecar during deployement (private key, public key, authority cert...), we wouldn't have add extra config parameters to our apps, etc...